### DISCUSSION TITLE
Activate route calculation at the end

### DISCUSSION THREAD

**Discussion by Leeb64** [devemux86/cruiser#364] [Q&A] [answered]
When planning a route (with BRouter) where I know various intermediate destinations in advance, it would make sense to only activate the route calculation at the end, after all intermediate destinations have been entered.
Even if I want to add 5-6 shaping points to the finished planned and calculated daily route (~300km), it would be good to only start the route calculation after entering the SP.
So you always have to wait until a calculation is finished before you can enter the next SP.

**Comment by devemux86** [comment]
@Leeb64 thanks, this is an interesting suggestion.

> So you always have to wait until a calculation is finished before you can enter the next SP.

We can add / remove / move waypoints while the route calculation is running.
The old route calculation is canceled and a new route calculation starts again.

**Reply by Ciutnik** [reply]
To be precise: possibility of changing the route - i.e. add a via point or extend the route - we only have when the first (any) section of the route has already been calculated. So we always have to wait for this first calculation. 

I once had a similar idea to run the calculation after setting all the route points, but you explained that we would lose sight of the location of a possible router error. Therefore, it is better to calculate the route in sections, one by one, rather than all at once.

**Reply by Leeb64** [reply]
I enter route points into the finished route so that I can return to the originally planned route in the event of a new route calculation (detour). I can enter the route points without waiting for the calculation, but the order is then not correct!

**Reply by Ciutnik** [reply]
Maybe, to avoid confusion, this menu item (but only when starting route planning) should be renamed to "Set end / via point"?

<img src="https://github.com/devemux86/cruiser/assets/122647782/f8c9399d-c717-4290-a587-9ec416a1c54b" width="50%"/>

**Reply by devemux86** [reply]
@Ciutnik when there is no route, a new route with 2 waypoints always has start - end.
Otherwise users would expect the yellow numbered icons and could be confused.

After we extend a route, the end or start becomes via / shaping.

**Reply by Leeb64** [reply]
When entering intermediate points, this crosshair appears with an "x" and an "r".
Maybe add a "+" there?
And to calculate the route, when all RPs are set, you could use the navigation start button, which until then (without a calculated route) looks somehow different (red?).

**Comment by devemux86** [comment, marked as answer]
Implemented and discussed here:
- https://github.com/devemux86/cruiser/discussions/449
